### PR TITLE
Estimate the memory size of the index before building the index

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: Cai.Zhang <cai.zhang@zilliz.com>

issue: #10754

### CODE REVIEW

File: internal/indexcoord/index_coord.go [hunk 1]
```diff
@@ -816,7 +818,7 @@ func (i *IndexCoord) assignTaskLoop() {
 					continue
 				}
 				log.Debug("The version of the task has been updated", zap.Int64("indexBuildID", indexBuildID))
-				nodeID, builderClient := i.nodeManager.PeekClient()
+				nodeID, builderClient := i.nodeManager.PeekClient(meta.indexMeta.IndexSize * 4)
```
Comment (czs007) [inline internal/indexcoord/index_coord.go:821]: It is best to change to a form that can be configured with different policies. Please refer to datacoord's approach

Comment (czs007) [reply internal/indexcoord/index_coord.go:821]: No need to store IndexSize (estimated) here.
Please add a SerializeSize member to store the memory amount of serialized object.

Comment (xiaocai2333) [reply internal/indexcoord/index_coord.go]: Yeah，I added the policies for peeking client.

**Pull request closed without merging**